core/src/apps/cardano/helpers/paths.py

```diff
@@ -7,21 +7,21 @@
 
 BYRON_ROOT = [44 | HARDENED, SLIP44_ID | HARDENED]
 SHELLEY_ROOT = [1852 | HARDENED, SLIP44_ID | HARDENED]
+MULTISIG_ROOT = [1854 | HARDENED, SLIP44_ID | HARDENED]
+MINTING_ROOT = [1855 | HARDENED, SLIP44_ID | HARDENED]
 
 # fmt: off
-SCHEMA_PUBKEY = PathSchema.parse("m/[44,1852]'/coin_type'/account'/*", SLIP44_ID)
-SCHEMA_ADDRESS = PathSchema.parse("m/[44,1852]'/coin_type'/account'/[0,1,2]/address_index", SLIP44_ID)
+SCHEMA_PUBKEY = PathSchema.parse("m/[44,1852,1854]'/coin_type'/account'/*", SLIP44_ID)
+# minting has a specific schema for key derivation - see CIP-1855
+SCHEMA_MINT = PathSchema.parse("m/1855'/coin_type'/[0-%s]'" % (HARDENED - 1), SLIP44_ID)
+SCHEMA_PAYMENT = PathSchema.parse("m/[44,1852,1854]'/coin_type'/account'/[0,1]/address_index", SLIP44_ID)
 # staking is only allowed on Shelley paths with suffix /2/0
-SCHEMA_STAKING = PathSchema.parse("m/1852'/coin_type'/account'/2/0", SLIP44_ID)
-SCHEMA_STAKING_ANY_ACCOUNT = PathSchema.parse("m/1852'/coin_type'/[0-%s]'/2/0" % (HARDENED - 1), SLIP44_ID)
+SCHEMA_STAKING = PathSchema.parse("m/[1852,1854]'/coin_type'/account'/2/0", SLIP44_ID)
+SCHEMA_STAKING_ANY_ACCOUNT = PathSchema.parse("m/[1852,1854]'/coin_type'/[0-%s]'/2/0" % (HARDENED - 1), SLIP44_ID)
 # fmt: on
 
-# the maximum allowed change address.  this should be large enough for normal
-# use and still allow to quickly brute-force the correct bip32 path
-MAX_SAFE_CHANGE_ADDRESS_INDEX = const(1_000_000)
-MAX_SAFE_ACCOUNT_INDEX = const(100) | HARDENED
 ACCOUNT_PATH_INDEX = const(2)
-BIP_PATH_LENGTH = const(5)
+CHAIN_STAKING_KEY = 2
```

Comment:
Why not `const`?

Reply:
I didn't realise. https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/3aeb358c7113230bb281ecc397cf92be94891d9a